### PR TITLE
First structure for the package generator

### PR DESCRIPTION
**Pull request by hmunfru**
### REVIEWERS

@jesuspg  @flopezag 
### DESCRIPTION

This is the initial code for the inclusion of the software to generate Murano packages from current FIWARE infrastructure
### TESTING

Done locally

### CODE REVIEW

File: murano-apps/package-generator/tests/unit/settings/cookbooks_url [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[main]
+product: http://product.git
+child: http://child.git
+child2: http://child2.git
```
Comment (jesuspg) [inline murano-apps/package-generator/tests/unit/settings/cookbooks_url]: add new lines

File: murano-apps/package-generator/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+testresources
+python-keystoneclient==1.8.0
+python-glanceclient==1.1.0
+
+oslo.config>=2.3.0
+oslo.i18n>=1.5.0
+oslo.serialization>=1.6.0
+oslo.utils>=2.0.0
+git+https://github.com/openstack/murano
+python-muranoclient
```
Comment (jesuspg) [inline murano-apps/package-generator/requirements.txt]: versions?

File: package-generator/model/cookbook.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        itself plus all cookbooks of their cookbook children and
+        recursively.
+        :return: Cookbook array
+        """
+        cookbooks = []
+        for cookbook_child in self.cookbook_childs:
+            cookbooks.append(cookbook_child)
+            if len(cookbook_child.get_cookbooks_child()) != 0:
+                cookbooks_in = cookbook_child.get_all_cookbooks_child()
+                cookbooks.extend(x for x in cookbooks_in if not self._exists(x.name, cookbooks))
```
Comment (jesuspg) [inline package-generator/model/cookbook.py]: line too long

File: package-generator/model/product_package.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        cookbooks = []
+        cookbook = Cookbook(self.product.get_product_name(),
+                            self.product.is_enabler())
+        cookbooks.append(cookbook)
+
+        for cookbook_child in cookbook.get_cookbooks_child():
+            cookbooks.append(cookbook_child)
+            if len(cookbook_child.get_cookbooks_child()) != 0:
+                cookbooks_in = cookbook_child.get_all_cookbooks_child()
+                cookbooks.extend(x for x in cookbooks_in if not self._exists(x.name, cookbooks))
```
Comment (jesuspg) [inline package-generator/model/product_package.py]: line too long

File: package-generator/setup.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+# Get requirements list from requirements.txt file
+# > parse_requirements() returns generator of pip.req.InstallRequirement objects
+install_reqs = parse_requirements(REQUIREMENTS_FILE, session=False)
+# > requirements_list is a list of requirement; e.g. ['requests==2.6.0', 'Fabric==1.8.3']
+requirements_list = [str(ir.req) for ir in install_reqs]
+
+setup(name='package-generator',
+      version='0.0.1',
+      description='Package Generator',
+      url='https://github.com/telefonicaid/fiware-enablers/murano-apps/package-generator',
```
Comment (jesuspg) [inline package-generator/setup.py]: change url to new location

File: package-generator/model/cookbook.py [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+
+
+class Cookbook:
+    """This class represents the cookbook object.
+    """
+    def __init__(self, name, installator, enabler=False):
+        """
+        The constructor
+        :param name: cookbook name
+        :param cookbook_config:  configuration file for cookbook
```
Comment (flopezag) [inline package-generator/model/cookbook.py]: Do you mean installator?

**Comment by jesuspg** [comment]
LGTM

**Pull request merged**